lib/crawler/output_sink/elasticsearch.rb

```diff
@@ -33,13 +33,25 @@ def initialize(config)
         # initialize client now to fail fast if config is bad
         client
 
+        # ping ES by attempting to reach the index specified in config
+        ping_output_index
+
         @queue_lock = Mutex.new
         init_ingestion_stats
         system_logger.info(
           "Elasticsearch sink initialized for index [#{index_name}] with pipeline [#{pipeline}]"
         )
       end
 
+      def ping_output_index
+        # when the index is not found, indices.get() fails in an 'ugly' way.
+        # rescue, log, then raise a SystemExit to make termination more graceful
+        client.indices.get(index: config.output_index)
+      rescue StandardError
+        system_logger.info("Failed to find index #{config.output_index}, aborting")
+        raise SystemExit
```

Comment:
Also regarding `SystemExit`, I think it would be more verbose to create a custom error to raise. There are some already here: https://github.com/elastic/crawler/blob/main/lib/errors.rb